### PR TITLE
build: add asan and ubsan default options

### PR DESCRIPTION
**Pull request by ii14**
Use print_stacktrace=1 for UBSAN by default.

### CODE REVIEW

**Comment by bfredl** [comment]
how does this work if `UBSAN_OPTIONS` also is supplied? Will this be appended to the default string or completely override it?

**Comment by ii14** [comment]
`UBSAN_OPTIONS` is applied on top of the default options / appended

**Comment by bfredl** [comment]
in that case, next question: can we do the same with ASAN_OPTIONS? then `handle_abort=1,handle_sigill=1` could be builtin defaults, and only `log_path` has to be supplied in the env var.

**Comment by ii14** [comment]
Sure, we can change the defaults for ASAN too

File: src/nvim/main.c [hunk 1]
```diff
@@ -2222,3 +2222,17 @@ static void check_swap_exists_action(void)
...
 }
+
+#ifdef ENABLE_ASAN_UBSAN
+const char *__ubsan_default_options(void);
+const char *__ubsan_default_options(void)
+{
+  return "print_stacktrace=1";
+}
+
+const char *__asan_default_options(void);
```
Comment (justinmk) [inline src/nvim/main.c:2233]: These might fit in `assert.{h,c}` instead of main.c

**Pull request merged**